transformation/transformationPlugins/jsonMessageTransformers/jsonJSTransformer/src/test/java/org/opensearch/migrations/transform/JavascriptTransformerTest.java

```diff
@@ -46,4 +49,73 @@ private void testScriptPerformance(String script) throws Exception {
             }
         }
     }
+
+
+    @Test
+    public void testCustomMap() throws Exception {
+        class CustomMapAlwaysContainsKey extends HashMap<String, Object> {
+            @Override
+            public boolean containsKey(Object key) {
+                return true;
+            }
+        }
+
+        var input = new CustomMapAlwaysContainsKey();
+        var scriptCallsHas = "((ignored) => (map) => (map.has('missing')))";
+        try (var testTransformer = new JavascriptTransformer(scriptCallsHas, Map.of())) {
+            var result = testTransformer.transformJson(input);
+            Assertions.assertEquals(true, result);
+            Assertions.assertEquals(Boolean.class, result.getClass());
+        }
+
+        var scriptReturnsNewArray = "((ignored) => (map) => (['output']))";
+        try (var testTransformer = new JavascriptTransformer(scriptReturnsNewArray, Map.of())) {
+            var result = testTransformer.transformJson(input);
+            Assertions.assertEquals(List.of("output"), result);
+            Assertions.assertEquals("com.oracle.truffle.polyglot.PolyglotList", result.getClass().getName());
+        }
+
+        var scriptReturnsInputMap = "((ignored) => (map) => (map))";
+        try (var testTransformer = new JavascriptTransformer(scriptReturnsInputMap, Map.of())) {
+            var result = testTransformer.transformJson(input);
+            Assertions.assertEquals(input, result);
+            Assertions.assertEquals(CustomMapAlwaysContainsKey.class.getName(), result.getClass().getName());
+        }
+
+    }
+
+    @Test
+    public void testPromiseHandling() throws Exception {
+        // Script that returns a resolved Promise
+        var successScript =
+        "((ignored) => (map) => {" +
+            "return new Promise((resolve, reject) => {" +
+                "resolve(['output']);" +
+            "});" +
+        "});";
+
+        // Test success case
+        try (var testTransformer = new JavascriptTransformer(successScript, Map.of())) {
+            var result = testTransformer.transformJson(null);
+            Assertions.assertInstanceOf(List.class, result);
```

Comment:
nit - this case isn't in the above test.  I don't think that you need it in either, but the omission gave me a pause.